### PR TITLE
Add authorization to the app

### PR DESCRIPTION
**Pull request by Atifzada**
### Following changes have been made during this PR:

- [x]  A user can delete a post if it is theirs or if they have an admin role
- [x]  Add the "Delete" button to the view while making sure that only authorized users can see it.
- [x]  A user can delete a comment if it is theirs or if they have an admin role
- [x]  Add the "Delete" button to the view while making sure that only authorized users can see it.
- [x]  Use CanCanCan for the authorization.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    return unless user.present?
+
+    can :read, :all
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
+  end
+end
```
Comment (Gambit142) [inline app/models/ability.rb]: - [x] Great work creating this Ability class. However, the implementation based on the requirements is not complete. Currently, a normal user who doesn't have admin rights can not delete a post that was created by the user and the same goes for a comment that was created by the user. Kindly rewrite the code in the ability class to factor in this requirement 👍 💻. 

```ruby
class Ability
  include CanCan::Ability

  def initialize(user)
    can :read, Post, Public: true
    can :read, Comment, Public: true

    return unless user.present?

    can :read, :all

    # Allow regular users to manage their own posts and comments
    can :manage, Post, author_id: user.id
    can :manage, Comment, author_id: user.id

    return unless user.role == 'admin'

    can :manage, :all
  end
end
```

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Atifzada @UbdaNam 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Installed `cancancan` gem ✔️
- Added `role` column to `Users` table ✔️ 
- README file is professional ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/models/ability.rb (hunk 1 above)
Comment (Atifzada) [reply app/models/ability.rb]: Thank you for suggesting the missing requirements. Surely I will fix it

**Comment by Gambit142** [comment]
@Atifzada, you are welcome 👍

**Review by OLIPLICHE — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Add CanCanCan ability that defines the set of rules 👏
- Role column exists in the users table👏


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**